src/AppInstallerCommonCore/ExperimentalFeature.cpp

```diff
@@ -3,36 +3,51 @@
 #pragma once
 #include "pch.h"
 #include "winget/ExperimentalFeature.h"
+#include "winget/GroupPolicy.h"
 #include "winget/UserSettings.h"
 
 namespace AppInstaller::Settings
 {
     bool ExperimentalFeature::IsEnabled(Feature feature)
     {
-        switch (feature)
+        return IsEnabled(feature, User());
+    }
+
+    bool ExperimentalFeature::IsEnabled(Feature feature, const UserSettings& userSettings)
+    {
+        if (feature == Feature::None)
         {
-        case Feature::None:
             return true;
+        }
+
+        // TODO: How do we report that this was due to policy higher up?
+        if (!GroupPolicies().IsEnabledOrNotConfigured(TogglePolicy::ExperimentalFeatures))
```

Comment:
>if (!GroupPolicies().IsEnabledOrNotConfigured(TogglePolicy::ExperimentalFeatures)) [](start = 8, length = 82)

Can we have an `IsDisabled` method as well?  It will read much more clearly instead of "not is enabled or not configured".

Reply:
I can add it if you think it's a good idea.

I originally had a single `IsAllowed` but then I wanted to be more explicit because some policies need a distinction between not-configured and enabled (e.g. default source), but in others it means the same thing (e.g. settings). We currently don't have policies where not-configured and disabled mean the same but I don't discard that possibility in the future.

Reply:
I agree that the semantics are an issue, mostly because they could change in the future.  "IsEnabled" or "IsDisabled" implies some amount of "handle the not set case by knowing the default".  However, we need that decision to be centralized.  We don't want the caller deciding how to handle the unset state; it could lead to bugs.

Given that we need that control to be centralized, I think that `IsEnabled` is the best choice of action, and we would have to encode the default state into a type if it is ever disabled.

---
In reply to: [598018838](https://github.com/microsoft/winget-cli/pull/810#discussion_r598018838) [](ancestors = 598018838)

Reply:
Okay. I had a function like that and a member like `TogglePolicyInternal::m_defaultIsEnabled` to guide it; I'll go back to that. And for the few places where we care about whether it was by default or explicit I'll use the enum I had for the state.